pkg/rbac/common.go

```
@@ -279,3 +300,202 @@ func isRuleInTargetAPIGroup(rule rbacv1.PolicyRule) bool {
 	}
 	return false
 }
+
+// CreateOrUpdateResource creates or updates the given resource
+//   - getResource is a func that returns a single object and an error
+//   - obj is the resource to create or update
+//   - client is the Wrangler client to use to get/create/update resource
+//   - areResourcesTheSame is a func that compares two resources and returns (true, nil) if they are equal, and (false, T) when not the same where T is an updated resource
+func CreateOrUpdateResource[T generic.RuntimeMetaObject, TList runtime.Object](obj T, client generic.NonNamespacedClientInterface[T, TList], areResourcesTheSame func(T, T) (bool, T)) error {
+	// attempt to get the resource
+	resource, err := client.Get(obj.GetName(), metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil
```

should we return the error here? Imagine there is an error getting the resource (e.g. cluster unavailable) if we return nil we would not retry.

---

Addressed in b6c18a7a080b2d7bfef6fb70efa0adaa74034a2c